### PR TITLE
[BEAM-7277] Add PostCommit test suite for Python 3.7

### PR DESCRIPTION
**Pull request by fredo838**
This is is part of a series of PRs with goal to make Apache Beam PY3 compatible. The proposal with the outlined approach has been documented here: https://s.apache.org/beam-python-3.

This PR adds a PostCommit test suite for Python  3.7. 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by fredo838** [comment]
@tvalentyn Is this what you need? I am not familiar with gradle.

**Comment by fredo838** [comment]
Run Python PostCommit

File: sdks/python/test-suites/direct/py37/build.gradle [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins { id 'org.apache.beam.module' }
+applyPythonNature()
+
+// Required to setup a Python 3 virtualenv.
+pythonVersion = '3.6'
```
Comment (tvalentyn) [inline sdks/python/test-suites/direct/py37/build.gradle]: 3.7

**Comment by tvalentyn** [comment]
Run Python Dataflow ValidatesRunner

**Comment by tvalentyn** [comment]
Also, we need to trigger these suites in a few places in this file: 
https://github.com/apache/beam/blob/cbc2618895622ba5edc2b021b207c600e868871f/build.gradle#L201

**Comment by tvalentyn** [comment]
Otherwise Jenkins does not trigger these suites, as we can see from current Postcommit test run on this PR.

**Comment by tvalentyn** [comment]
Postcommit tests failed due to the error: 
"09:10:20 root: INFO: 2019-05-13T16:02:15.972Z: JOB_MESSAGE_ERROR: Startup of the worker pool in zone us-central1-a failed to bring up any of the desired 1 workers. QUOTA_EXCEEDED: Quota 'DISKS_TOTAL_GB' exceeded.  Limit: 200000.0 in region us-central1.". This is an environmental issue, and I sent a request to get more quota.

**Comment by fredo838** [comment]
Run Python PostCommit

**Comment by tvalentyn** [comment]
There is an error:
```
12:01:44 Could not determine the dependencies of task ':python3PostCommit'.
12:01:44 > Task with path ':beam-sdks-python-test-suites-dataflow-py37:postCommitIT' not found in root project 'beam
```

**Comment by fredo838** [comment]
Run Python PreCommit

**Comment by fredo838** [comment]
@tvalentyn PTAL

**Comment by tvalentyn** [comment]
Looks like there is a merge conflict, could you please rebase on top of master?

**Comment by tvalentyn** [comment]
Once that's completed, please trigger the postcommit suite.

**Comment by fredo838** [comment]
Run Python PostCommit

File: settings.gradle [hunk 2]
```diff
@@ -48,98 +48,190 @@ include ":runners:flink:1.8"
...
-include ":vendor:grpc-1_13_1"
-include ":sdks:java:testing:test-utils"
-include ":vendor:sdks-java-extensions-protobuf"
-include ":vendor:guava-20_0"
-include ":website"
-include ":runners:google-cloud-dataflow-java:worker:legacy-worker"
-include ":runners:google-cloud-dataflow-java:worker"
-include ":runners:google-cloud-dataflow-java:worker:windmill"
-// no dots allowed for project paths
+include "beam-runners-gearpump"
```
Comment (tvalentyn) [inline settings.gradle]: I think changes to settings.gradle on this PR are unintentional?

**Comment by fredo838** [comment]
Run Python PostCommit

**Comment by fredo838** [comment]
I got following error: `Project with path ':beam-runners-google-cloud-dataflow-java-fn-api-worker' could not be found in project ':sdks:python:test-suites:dataflow:py37'.`

**Comment by tvalentyn** [comment]
Run Python PostCommit

**Comment by tvalentyn** [comment]
There is still: ":sdks:python:test-suites:direct:py37:postCommitIT' not found in root project 'beam'."

**Comment by tvalentyn** [comment]
Also please add the datastore test to the Direct runner suites on 3.6, 3.7. Thanks!

**Comment by tvalentyn** [comment]
@pabloem could you please help with the merge? Thank you!

File: settings.gradle [hunk 3]
```diff
@@ -126,8 +126,10 @@ include ":sdks:python:container:py3"
 include ":sdks:python:test-suites:dataflow"
 include ":sdks:python:test-suites:dataflow:py35"
 include ":sdks:python:test-suites:dataflow:py36"
+include ":sdks:python:test-suites:dataflow:py37"
 include ":sdks:python:test-suites:direct:py35"
 include ":sdks:python:test-suites:direct:py36"
+include ":sdks:python:test-suites:direct:py36"
```
Comment (udim) [inline settings.gradle]: I believe a typo here is breaking python postcommits: should be `py37`?

Comment (udim) [reply settings.gradle]: Example log: https://builds.apache.org/job/beam_PostCommit_Python3_Verify_PR/321/console

Comment (tvalentyn) [reply settings.gradle]: Thank you, @udim. Sent https://github.com/apache/beam/pull/8601. For some reason the error didn't happen in postcommit run on this PR.

Comment (tvalentyn) [reply settings.gradle]: Look like most recent test run on this PR did not include postcommits, I overlooked it.

**Pull request merged**